Maven 3.8.1 update causes warning

Maven was recently updated from 3.6.3 to 3.8.1 to address [a security vulnerability](https://maven.apache.org/docs/3.8.1/release-notes.html).  As a result, this warning now comes up on startup...
```
2021-04-04 19:02:43,763 WARN  [io.qua.boo.res.mav.BootstrapMavenContext] (main) Settings problem encountered at /Users/francois/Libraries/Java/apache-maven/conf/settings.xml, line 164, column 16: org.apache.maven.settings.io.SettingsParseException: Unrecognised tag: 'blocked' (position: START_TAG seen ...</url>\r\n      <blocked>... @164:16) 
        at org.apache.maven.settings.io.DefaultSettingsReader.read(DefaultSettingsReader.java:87)
        at org.apache.maven.settings.building.DefaultSettingsBuilder.readSettings(DefaultSettingsBuilder.java:182)
        at org.apache.maven.settings.building.DefaultSettingsBuilder.build(DefaultSettingsBuilder.java:104)
        at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.getEffectiveSettings(BootstrapMavenContext.java:252)
        at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.isOffline(BootstrapMavenContext.java:218)
        at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.newRepositorySystem(BootstrapMavenContext.java:648)
        at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.getRepositorySystem(BootstrapMavenContext.java:223)
        at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.<init>(MavenArtifactResolver.java:125)
        at io.quarkus.bootstrap.BootstrapAppModelFactory.getAppModelResolver(BootstrapAppModelFactory.java:192)
        at io.quarkus.bootstrap.BootstrapAppModelFactory.resolveAppModel(BootstrapAppModelFactory.java:305)
        at io.quarkus.bootstrap.app.QuarkusBootstrap.bootstrap(QuarkusBootstrap.java:163)
        at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:143)
        at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:63)
Caused by: org.codehaus.plexus.util.xml.pull.XmlPullParserException: Unrecognised tag: 'blocked' (position: START_TAG seen ...</url>\r\n      <blocked>... @164:16) 
        at org.apache.maven.settings.io.xpp3.SettingsXpp3Reader.checkUnknownElement(SettingsXpp3Reader.java:149)
        at org.apache.maven.settings.io.xpp3.SettingsXpp3Reader.parseMirror(SettingsXpp3Reader.java:920)
        at org.apache.maven.settings.io.xpp3.SettingsXpp3Reader.parseSettings(SettingsXpp3Reader.java:1416)
        at org.apache.maven.settings.io.xpp3.SettingsXpp3Reader.read(SettingsXpp3Reader.java:1544)
        at org.apache.maven.settings.io.xpp3.SettingsXpp3Reader.read(SettingsXpp3Reader.java:566)
        at org.apache.maven.settings.io.xpp3.SettingsXpp3Reader.read(SettingsXpp3Reader.java:595)
        at org.apache.maven.settings.io.DefaultSettingsReader.read(DefaultSettingsReader.java:83)
        ... 12 more
```

Quarkus 1.12.1
Java 16

--- a/.github/workflows/ci-actions-incremental.yml
+++ b/.github/workflows/ci-actions-incremental.yml
@@ -172,7 +172,7 @@ jobs:
     needs: build-jdk11
     timeout-minutes: 240
     env:
-      MAVEN_OPTS: -Xmx2048m -XX:MaxMetaspaceSize=1000m
+      MAVEN_OPTS: -Xmx2g -XX:MaxMetaspaceSize=1g
     strategy:
       fail-fast: false
       matrix:
@@ -253,7 +253,7 @@ jobs:
     needs: build-jdk11
     timeout-minutes: 180
     env:
-      MAVEN_OPTS: -Xmx1408m  -XX:MaxMetaspaceSize=1000m
+      MAVEN_OPTS: -Xmx1500m -XX:MaxMetaspaceSize=1g
 
     steps:
       - uses: actions/checkout@v2
@@ -301,6 +301,8 @@ jobs:
   linux-jvm-maven-tests:
     name: Maven Tests - JDK ${{matrix.java.name}}
     runs-on: ubuntu-latest
+    env:
+      MAVEN_OPTS: -Xmx2g -XX:MaxMetaspaceSize=1g
     # Skip main in forks
     if: "github.repository == 'quarkusio/quarkus' || !endsWith(github.ref, '/main')"
     needs: build-jdk11
@@ -356,6 +358,8 @@ jobs:
   windows-jdk11-jvm-maven-tests:
     name: Maven Tests - JDK 11 Windows
     runs-on: windows-latest
+    env:
+      MAVEN_OPTS: -Xmx2g -XX:MaxMetaspaceSize=1g
     # Skip main in forks
     if: "github.repository == 'quarkusio/quarkus' || !endsWith(github.ref, '/main')"
     needs: build-jdk11
@@ -406,6 +410,9 @@ jobs:
   gradle-tests-jdk11-jvm:
     name: Gradle Tests - JDK 11 ${{matrix.os.family}}
     runs-on: ${{matrix.os.name}}
+    env:
+      # leave more space for the actual gradle execution (which is just wrapped by maven)
+      MAVEN_OPTS: -Xmx1g
     # Skip main in forks
     if: "github.repository == 'quarkusio/quarkus' || !endsWith(github.ref, '/main')"
     needs: build-jdk11
@@ -657,6 +664,9 @@ jobs:
     name: Native Tests - ${{matrix.category}}
     needs: [build-jdk11, native-tests-read-json-matrix]
     runs-on: ubuntu-latest
+    env:
+      # leave more space for the actual native compilation and execution
+      MAVEN_OPTS: -Xmx1g
     # Skip main in forks
     if: "github.repository == 'quarkusio/quarkus' || !endsWith(github.ref, '/main')"
     # Ignore the following YAML Schema error
@@ -727,6 +737,9 @@ jobs:
     name: Native Tests - Windows - ${{matrix.category}}
     needs: [build-jdk11, native-tests-read-json-matrix]
     runs-on: windows-latest
+    env:
+      # leave more space for the actual native compilation and execution
+      MAVEN_OPTS: -Xmx1g
     # Skip main in forks
     if: "github.repository == 'quarkusio/quarkus' || !endsWith(github.ref, '/main')"
     # Ignore the following YAML Schema error

--- a/.github/workflows/ci-actions-incremental.yml
+++ b/.github/workflows/ci-actions-incremental.yml
@@ -114,7 +114,7 @@ jobs:
         # note: '-Pincremental -Dgib.disable' shall only trigger the download of gitflow-incremental-builder, not activate it
         #       since -Dincremental deactivates dependencies in bom-descriptor-json (which we do need here), -P is used instead
         run: |
-          mvn -T1C $COMMON_MAVEN_ARGS -DskipTests -DskipITs -Dinvoker.skip -Dno-format -Dtcks -Pincremental -Dgib.disable clean install
+          ./mvnw -T1C $COMMON_MAVEN_ARGS -DskipTests -DskipITs -Dinvoker.skip -Dno-format -Dtcks -Pincremental -Dgib.disable clean install
       - name: Verify extension dependencies
         shell: bash
         run: |
@@ -226,7 +226,7 @@ jobs:
         run: tar -xzf maven-repo.tgz -C ~
       - name: Build
         shell: bash
-        run: mvn $COMMON_MAVEN_ARGS $JVM_TEST_MAVEN_OPTS install -Dsurefire.timeout=600 -pl !integration-tests/gradle -pl !integration-tests/maven -pl !integration-tests/devtools ${{ matrix.java.maven_args }} ${{ needs.build-jdk11.outputs.gib_args }}
+        run: ./mvnw $COMMON_MAVEN_ARGS $JVM_TEST_MAVEN_OPTS install -Dsurefire.timeout=600 -pl !integration-tests/gradle -pl !integration-tests/maven -pl !integration-tests/devtools ${{ matrix.java.maven_args }} ${{ needs.build-jdk11.outputs.gib_args }}
       - name: Prepare failure archive (if maven failed)
         if: failure()
         shell: bash
@@ -277,7 +277,7 @@ jobs:
         run: tar -xzf maven-repo.tgz -C ~
       - name: Build
         shell: bash
-        run: mvn $COMMON_MAVEN_ARGS -DskipDocs -Dformat.skip -Dsurefire.timeout=600 -pl !integration-tests/gradle -pl !integration-tests/maven -pl !integration-tests/devtools install ${{ needs.build-jdk11.outputs.gib_args }}
+        run: ./mvnw $COMMON_MAVEN_ARGS -DskipDocs -Dformat.skip -Dsurefire.timeout=600 -pl !integration-tests/gradle -pl !integration-tests/maven -pl !integration-tests/devtools install ${{ needs.build-jdk11.outputs.gib_args }}
       - name: Prepare failure archive (if maven failed)
         if: failure()
         shell: bash
@@ -334,7 +334,7 @@ jobs:
         with:
           java-version: ${{ matrix.java.java-version }}
       - name: Build
-        run: mvn $COMMON_MAVEN_ARGS $JVM_TEST_MAVEN_OPTS install -pl 'integration-tests/maven' ${{ needs.build-jdk11.outputs.gib_args }}
+        run: ./mvnw $COMMON_MAVEN_ARGS $JVM_TEST_MAVEN_OPTS install -pl 'integration-tests/maven' ${{ needs.build-jdk11.outputs.gib_args }}
       - name: Prepare failure archive (if maven failed)
         if: failure()
         shell: bash
@@ -384,7 +384,7 @@ jobs:
           java-version: 11
       - name: Build
         shell: bash
-        run: mvn $COMMON_MAVEN_ARGS $JVM_TEST_MAVEN_OPTS install -pl 'integration-tests/maven' ${{ needs.build-jdk11.outputs.gib_args }}
+        run: ./mvnw $COMMON_MAVEN_ARGS $JVM_TEST_MAVEN_OPTS install -pl 'integration-tests/maven' ${{ needs.build-jdk11.outputs.gib_args }}
       - name: Prepare failure archive (if maven failed)
         if: failure()
         shell: bash
@@ -461,7 +461,7 @@ jobs:
         working-directory: integration-tests/gradle
       - name: Build
         shell: bash
-        run: mvn $COMMON_MAVEN_ARGS $JVM_TEST_MAVEN_OPTS install -pl integration-tests/gradle ${{ needs.build-jdk11.outputs.gib_args }}
+        run: ./mvnw $COMMON_MAVEN_ARGS $JVM_TEST_MAVEN_OPTS install -pl integration-tests/gradle ${{ needs.build-jdk11.outputs.gib_args }}
       - name: Upload Surefire reports (if build failed)
         uses: actions/upload-artifact@v2
         if: failure()
@@ -506,7 +506,7 @@ jobs:
         with:
           java-version: ${{ matrix.java.java-version }}
       - name: Build
-        run: mvn $COMMON_MAVEN_ARGS $JVM_TEST_MAVEN_OPTS install -pl 'integration-tests/devtools' ${{ needs.build-jdk11.outputs.gib_args }}
+        run: ./mvnw $COMMON_MAVEN_ARGS $JVM_TEST_MAVEN_OPTS install -pl 'integration-tests/devtools' ${{ needs.build-jdk11.outputs.gib_args }}
       - name: Prepare failure archive (if maven failed)
         if: failure()
         shell: bash
@@ -556,7 +556,7 @@ jobs:
           java-version: 11
       - name: Build
         shell: bash
-        run: mvn $COMMON_MAVEN_ARGS $JVM_TEST_MAVEN_OPTS install -pl 'integration-tests/devtools' ${{ needs.build-jdk11.outputs.gib_args }}
+        run: ./mvnw $COMMON_MAVEN_ARGS $JVM_TEST_MAVEN_OPTS install -pl 'integration-tests/devtools' ${{ needs.build-jdk11.outputs.gib_args }}
       - name: Prepare failure archive (if maven failed)
         if: failure()
         shell: bash
@@ -618,7 +618,7 @@ jobs:
         run: mvn -B --settings ../tcks/.github/mvn-settings.xml install
         working-directory: ./resteasy-reactive-testsuite
       - name: Verify with Maven
-        run: mvn $COMMON_MAVEN_ARGS -Dtcks -pl tcks -amd install ${{ needs.build-jdk11.outputs.gib_args }}
+        run: ./mvnw $COMMON_MAVEN_ARGS -Dtcks -pl tcks -amd install ${{ needs.build-jdk11.outputs.gib_args }}
         working-directory: ./tcks
       - name: Prepare failure archive (if maven failed)
         if: failure()
@@ -699,12 +699,12 @@ jobs:
         run: |
           for i in $TEST_MODULES
           do modules+="integration-tests/$i,"; done
-          mvn $COMMON_MAVEN_ARGS -pl "${modules}" $NATIVE_TEST_MAVEN_OPTS ${{ needs.build-jdk11.outputs.gib_args }}
+          ./mvnw $COMMON_MAVEN_ARGS -pl "${modules}" $NATIVE_TEST_MAVEN_OPTS ${{ needs.build-jdk11.outputs.gib_args }}
           # add the 'simple with spaces' project to the run of 'Misc1' by executing it explicitly
           # done because there is no good way to pass strings with empty values to the previous command
           # so this hack is as good as any
           if [ "$CATEGORY" == "Misc1" ]; then
-            mvn $COMMON_MAVEN_ARGS -Dnative -Dquarkus.native.container-build=true -pl 'integration-tests/simple with space/' verify ${{ needs.build-jdk11.outputs.gib_args }}
+            ./mvnw $COMMON_MAVEN_ARGS -Dnative -Dquarkus.native.container-build=true -pl 'integration-tests/simple with space/' verify ${{ needs.build-jdk11.outputs.gib_args }}
           fi
       - name: Prepare failure archive (if maven failed)
         if: failure()
@@ -789,7 +789,7 @@ jobs:
         run: |
           for i in $TEST_MODULES
           do modules+="integration-tests/$i,"; done
-          mvn $COMMON_MAVEN_ARGS -pl "${modules}" $WINDOWS_NATIVE_TEST_MAVEN_OPTS ${{ needs.build-jdk11.outputs.gib_args }}
+          ./mvnw $COMMON_MAVEN_ARGS -pl "${modules}" $WINDOWS_NATIVE_TEST_MAVEN_OPTS ${{ needs.build-jdk11.outputs.gib_args }}
       - name: Prepare failure archive (if maven failed)
         if: failure()
         shell: bash

--- a/.github/workflows/ci-actions.yml.disabled
+++ b/.github/workflows/ci-actions.yml.disabled
@@ -91,7 +91,7 @@ jobs:
           key: q2maven-${{ steps.get-date.outputs.date }}
       - name: Build
         run: |
-          mvn -T1C -e -B -DskipTests -DskipITs -Dinvoker.skip -Dno-format -Dtcks --settings .github/mvn-settings.xml clean install
+          ./mvnw -T1C -e -B -DskipTests -DskipITs -Dinvoker.skip -Dno-format -Dtcks --settings .github/mvn-settings.xml clean install
       - name: Verify extension dependencies
         shell: bash
         run: |
@@ -173,7 +173,7 @@ jobs:
         shell: bash
         run: tar -xzf maven-repo.tgz -C ~
       - name: Build with Maven
-        run: eval mvn $JVM_TEST_MAVEN_OPTS install -pl !integration-tests/gradle -pl !integration-tests/maven -pl !integration-tests/devtools ${{ matrix.java.maven_args}}
+        run: eval ./mvnw $JVM_TEST_MAVEN_OPTS install -pl !integration-tests/gradle -pl !integration-tests/maven -pl !integration-tests/devtools ${{ matrix.java.maven_args}}
       - name: Prepare failure archive (if maven failed)
         if: failure()
         shell: bash
@@ -212,7 +212,7 @@ jobs:
         run: tar -xzf maven-repo.tgz -C ~
       - name: Build with Maven
         shell: bash
-        run: mvn -B --settings .github/mvn-settings.xml -DskipDocs -Dformat.skip -pl !integration-tests/gradle -pl !integration-tests/maven -pl !integration-tests/devtools install
+        run: ./mvnw -B --settings .github/mvn-settings.xml -DskipDocs -Dformat.skip -pl !integration-tests/gradle -pl !integration-tests/maven -pl !integration-tests/devtools install
       - name: Prepare failure archive (if maven failed)
         if: failure()
         shell: bash
@@ -257,7 +257,7 @@ jobs:
         with:
           java-version: ${{ matrix.java.java-version }}
       - name: Run Maven integration tests
-        run: eval mvn $JVM_TEST_MAVEN_OPTS install -pl 'integration-tests/maven'
+        run: eval ./mvnw $JVM_TEST_MAVEN_OPTS install -pl 'integration-tests/maven'
       - name: Prepare failure archive (if maven failed)
         if: failure()
         shell: bash
@@ -295,7 +295,7 @@ jobs:
           java-version: 11
       - name: Run Maven integration tests
         shell: bash
-        run: mvn $JVM_TEST_MAVEN_OPTS install -pl 'integration-tests/maven'
+        run: ./mvnw $JVM_TEST_MAVEN_OPTS install -pl 'integration-tests/maven'
       - name: Prepare failure archive (if maven failed)
         if: failure()
         shell: bash
@@ -411,7 +411,7 @@ jobs:
         with:
           java-version: ${{ matrix.java.java-version }}
       - name: Run Devtools integration tests
-        run: eval mvn $JVM_TEST_MAVEN_OPTS install -pl 'integration-tests/devtools'
+        run: eval ./mvnw $JVM_TEST_MAVEN_OPTS install -pl 'integration-tests/devtools'
       - name: Prepare failure archive (if maven failed)
         if: failure()
         shell: bash
@@ -449,7 +449,7 @@ jobs:
           java-version: 11
       - name: Run Devtools integration tests
         shell: bash
-        run: mvn $JVM_TEST_MAVEN_OPTS install -pl 'integration-tests/devtools'
+        run: ./mvnw $JVM_TEST_MAVEN_OPTS install -pl 'integration-tests/devtools'
       - name: Prepare failure archive (if maven failed)
         if: failure()
         shell: bash
@@ -498,7 +498,7 @@ jobs:
         run: mvn -B --settings ../tcks/.github/mvn-settings.xml install
         working-directory: ./resteasy-reactive-testsuite
       - name: Verify with Maven
-        run: mvn -B --settings .github/mvn-settings.xml -f tcks/pom.xml install
+        run: ../mvnw -B --settings .github/mvn-settings.xml -f tcks/pom.xml install
         working-directory: ./tcks
       - name: Prepare failure archive (if maven failed)
         if: failure()
@@ -568,12 +568,12 @@ jobs:
           for i in $TEST_MODULES
           do modules+=("integration-tests/$i"); done
           IFS=,
-          eval mvn -pl "${modules[*]}" $NATIVE_TEST_MAVEN_OPTS
+          eval ./mvnw -pl "${modules[*]}" $NATIVE_TEST_MAVEN_OPTS
           # add the 'simple with spaces' project to the run of 'Misc1' by executing it explicitly
           # done because there is no good way to pass strings with empty values to the previous command
           # so this hack is as good as any
           if [ "$CATEGORY" == "Misc1" ]; then
-            mvn -Dnative -Dquarkus.native.container-build=true -B --settings .github/mvn-settings.xml -f 'integration-tests/simple with space/' verify
+            ./mvnw -Dnative -Dquarkus.native.container-build=true -B --settings .github/mvn-settings.xml -f 'integration-tests/simple with space/' verify
           fi
       - name: Prepare failure archive (if maven failed)
         if: failure()

--- a/.github/workflows/ci-fork-mvn-cache.yml
+++ b/.github/workflows/ci-fork-mvn-cache.yml
@@ -55,7 +55,7 @@ jobs:
           key: q2maven-${{ steps.get-date.outputs.date }}
       - name: Build
         run: |
-          mvn -T1C -e -B --settings .github/mvn-settings.xml -Dquickly-ci -Dtcks clean install
+          ./mvnw -T1C -e -B --settings .github/mvn-settings.xml -Dquickly-ci -Dtcks clean install
       - name: Delete Local Artifacts From Cache
         shell: bash
         run: rm -r ~/.m2/repository/io/quarkus

--- a/.github/workflows/codeql-analysis.yml
+++ b/.github/workflows/codeql-analysis.yml
@@ -51,7 +51,7 @@ jobs:
 
     - if: matrix.language == 'java' 
       name: Build Java
-      run: mvn -B --settings .github/mvn-settings.xml -Dquickly-ci install
+      run: ./mvnw -B --settings .github/mvn-settings.xml -Dquickly-ci install
 
     - name: Perform CodeQL Analysis
       uses: github/codeql-action/analyze@v1

--- a/.github/workflows/deploy-snapshots.yml
+++ b/.github/workflows/deploy-snapshots.yml
@@ -43,7 +43,7 @@ jobs:
           SERVER_USERNAME: ${{ secrets.SERVER_USERNAME }}
           SERVER_PASSWORD: ${{ secrets.SERVER_PASSWORD }}
         run: |
-          mvn -e -B --settings .github/mvn-settings.xml \
+          ./mvnw -e -B --settings .github/mvn-settings.xml \
             -DskipITs -Dno-format -Dinvoker.skip=true \
             -DretryFailedDeploymentCount=10 \
             clean deploy

--- a/.github/workflows/doc-build.yml
+++ b/.github/workflows/doc-build.yml
@@ -46,7 +46,7 @@ jobs:
           key: q2maven-doc-${{ steps.get-date.outputs.date }}
       - name: Build
         run: |
-          mvn -Dquickly-ci -B --settings .github/mvn-settings.xml install
+          ./mvnw -Dquickly-ci -B --settings .github/mvn-settings.xml install
       - name: Build Docs
         run: |
-          mvn -e -B --settings .github/mvn-settings.xml clean org.asciidoctor:asciidoctor-maven-plugin:process-asciidoc -pl docs -Ddocumentation-pdf
+          ./mvnw -e -B --settings .github/mvn-settings.xml clean org.asciidoctor:asciidoctor-maven-plugin:process-asciidoc -pl docs -Ddocumentation-pdf

--- a/.github/workflows/jdk-early-access-build.yml
+++ b/.github/workflows/jdk-early-access-build.yml
@@ -71,7 +71,7 @@ jobs:
         # -fae to try to gather as many failures as possible
         # (but not maven.test.failure.ignore because test report generation is buggy)
         run: |
-          mvn $JVM_TEST_MAVEN_OPTS -Dtcks install -fae
+          ./mvnw $JVM_TEST_MAVEN_OPTS -Dtcks install -fae
       # Test reports disabled due to: https://github.com/ScaCap/action-surefire-report/issues/39
       #- name: Publish Test Report
       #  if: always()

--- a/.github/workflows/native-cron-build.yml.disabled
+++ b/.github/workflows/native-cron-build.yml.disabled
@@ -52,10 +52,10 @@ jobs:
             ${{ runner.os }}-native-cron-
 
       - name: Build Quarkus
-        run: mvn -B install -DskipTests -DskipITs -Dformat.skip
+        run: ./mvnw -B install -DskipTests -DskipITs -Dformat.skip
 
       - name: Run integration tests in native
-        run: mvn -B --settings .github/mvn-settings.xml verify -f integration-tests/pom.xml --fail-at-end -Dno-format -Dtest-containers -Dstart-containers -Dnative -Dquarkus.native.container-build=true -Dquarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-native-image:20.2.0-java${{ matrix.java }} -pl '!io.quarkus:quarkus-integration-test-google-cloud-functions-http,!io.quarkus:quarkus-integration-test-google-cloud-functions,!io.quarkus:quarkus-integration-test-funqy-google-cloud-functions'
+        run: ./mvnw -B --settings .github/mvn-settings.xml verify -f integration-tests/pom.xml --fail-at-end -Dno-format -Dtest-containers -Dstart-containers -Dnative -Dquarkus.native.container-build=true -Dquarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-native-image:20.2.0-java${{ matrix.java }} -pl '!io.quarkus:quarkus-integration-test-google-cloud-functions-http,!io.quarkus:quarkus-integration-test-google-cloud-functions,!io.quarkus:quarkus-integration-test-funqy-google-cloud-functions'
 
       - name: Report
         if: always()

--- a/.github/workflows/release-build.yml
+++ b/.github/workflows/release-build.yml
@@ -27,7 +27,7 @@ jobs:
         run: mkdir -p $HOME/release/repository
       - name: Build and Test
         run: |
-          mvn --settings .github/mvn-settings.xml \
+          ./mvnw --settings .github/mvn-settings.xml \
             -B \
             -DskipITs -Dinvoker.skip=true \
             -Dmaven.repo.local=$HOME/release/repository \

--- a/.mvn/wrapper/MavenWrapperDownloader.java
+++ b/.mvn/wrapper/MavenWrapperDownloader.java
@@ -5,7 +5,7 @@
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
- *      https://www.apache.org/licenses/LICENSE-2.0
+ *      http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,

--- a/.mvn/wrapper/maven-wrapper.properties
+++ b/.mvn/wrapper/maven-wrapper.properties
@@ -1,2 +1,2 @@
-distributionUrl=https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/3.6.3/apache-maven-3.6.3-bin.zip
+distributionUrl=https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/3.8.1/apache-maven-3.8.1-bin.zip
 wrapperUrl=https://repo.maven.apache.org/maven2/io/takari/maven-wrapper/0.5.6/maven-wrapper-0.5.6.jar

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -78,8 +78,8 @@
         <asm.version>9.1</asm.version>
         <commons-io.version>2.8.0</commons-io.version>
         <jboss-metadata-web.version>11.0.0.Final</jboss-metadata-web.version>
-        <maven-core.version>3.6.3</maven-core.version>
-        <maven-resolver.version>1.4.1</maven-resolver.version>
+        <maven-core.version>3.8.1</maven-core.version>
+        <maven-resolver.version>1.6.2</maven-resolver.version>
         <maven-toolchain.version>3.0-alpha-2</maven-toolchain.version>
         <guava.version>30.1.1-jre</guava.version>
         <plexus-utils.version>3.2.1</plexus-utils.version>

--- a/build-parent/pom.xml
+++ b/build-parent/pom.xml
@@ -55,7 +55,7 @@
         <supported-maven-versions>[3.6.2,)</supported-maven-versions>
 
         <!-- These 2 properties are used by CreateProjectMojo to add the Maven Wrapper -->
-        <proposed-maven-version>3.6.3</proposed-maven-version>
+        <proposed-maven-version>3.8.1</proposed-maven-version>
         <maven-wrapper.version>0.7.7</maven-wrapper.version>
         <gradle-wrapper.version>6.8.3</gradle-wrapper.version>
 
@@ -83,7 +83,7 @@
         <!-- Align various dependencies that are not really part of the bom-->
         <findbugs.version>3.0.2</findbugs.version>
         <jsoup.version>1.11.3</jsoup.version>
-        <wagon-provider-api.version>3.3.3</wagon-provider-api.version>
+        <wagon-provider-api.version>3.4.3</wagon-provider-api.version>
 
         <!-- The image to use for tests that run Keycloak -->
         <!-- IMPORTANT: If this is changed you must also update bom/application/pom.xml to match the version -->

--- a/build-parent/pom.xml
+++ b/build-parent/pom.xml
@@ -312,7 +312,7 @@
                             <maven.settings>${session.request.userSettingsFile.path}</maven.settings>
                             <project.version>${project.version}</project.version> <!-- some dev tools tests need this -->
                         </systemPropertyVariables>
-                        <argLine>${jacoco.agent.argLine} -Xmx1500m</argLine> <!-- limit the amount of memory surefire can use, 1500m should be plenty-->
+                        <argLine>${jacoco.agent.argLine} -Xmx1500m -XX:MaxMetaspaceSize=1500m</argLine> <!-- limit the amount of memory surefire can use, 1500m should be plenty-->
                         <!-- https://lists.apache.org/thread.html/r9030808273c82ac6d7b9602d34d446c7d8c4e8aa02c41bca164df1c5%40%3Cdev.maven.apache.org%3E -->
                         <trimStackTrace>false</trimStackTrace>
                     </configuration>

--- a/independent-projects/arc/pom.xml
+++ b/independent-projects/arc/pom.xml
@@ -40,7 +40,7 @@
         <version.jta>1.3.3</version.jta>
         <version.jandex>2.2.3.Final</version.jandex>
         <version.junit5>5.7.1</version.junit5>
-        <version.maven>3.6.3</version.maven>
+        <version.maven>3.8.1</version.maven>
         <version.assertj>3.19.0</version.assertj>
         <version.jboss-logging>3.4.1.Final</version.jboss-logging>
         <version.jakarta-annotation>1.3.5</version.jakarta-annotation>

--- a/independent-projects/bootstrap/pom.xml
+++ b/independent-projects/bootstrap/pom.xml
@@ -28,11 +28,12 @@
         <eclipse-minimal-json.version>0.9.5</eclipse-minimal-json.version>
         <jboss-logging.version>3.4.1.Final</jboss-logging.version>
         <junit.jupiter.version>5.7.1</junit.jupiter.version>
-        <maven-core.version>3.6.3</maven-core.version><!-- Keep in sync with sisu.version -->
+        <maven-core.version>3.8.1</maven-core.version><!-- Keep in sync with sisu.version -->
         <sisu.version>0.3.4</sisu.version><!-- Keep in sync with maven-core.version -->
         <maven-plugin-annotations.version>3.6.0</maven-plugin-annotations.version>
-        <maven-resolver.version>1.4.1</maven-resolver.version>
-        <maven-wagon.version>3.3.4</maven-wagon.version>
+        <maven-resolver.version>1.6.2</maven-resolver.version>
+        <maven-wagon.version>3.4.3</maven-wagon.version>
+        <httpcore.version>4.4.14</httpcore.version><!-- Keep in sync with maven-wagon.version (wagon-http 3.4.3 brings in 4.4.13 and 4.4.14) -->
         <plexus-cypher.version>1.7</plexus-cypher.version>
         <nexus-staging-maven-plugin.version>1.6.8</nexus-staging-maven-plugin.version>
         <commons-logging-jboss-logging.version>1.0.0.Final</commons-logging-jboss-logging.version>
@@ -292,6 +293,11 @@
                         <artifactId>commons-io</artifactId>
                     </exclusion>
                 </exclusions>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.httpcomponents</groupId>
+                <artifactId>httpcore</artifactId>
+                <version>${httpcore.version}</version>
             </dependency>
             <dependency>
                 <groupId>org.eclipse.sisu</groupId>

--- a/independent-projects/enforcer-rules/pom.xml
+++ b/independent-projects/enforcer-rules/pom.xml
@@ -28,7 +28,7 @@
 
         <enforcer-api.version>3.0.0-M3</enforcer-api.version>
         <maven-invoker-plugin.version>3.2.1</maven-invoker-plugin.version>
-        <maven-core.version>3.6.3</maven-core.version>
+        <maven-core.version>3.8.1</maven-core.version>
 
         <nexus-staging-maven-plugin.version>1.6.8</nexus-staging-maven-plugin.version>
     </properties>

--- a/independent-projects/resteasy-reactive/pom.xml
+++ b/independent-projects/resteasy-reactive/pom.xml
@@ -39,7 +39,7 @@
         <version.cdi>2.0.2</version.cdi>
         <version.jandex>2.2.3.Final</version.jandex>
         <version.junit5>5.7.1</version.junit5>
-        <version.maven>3.6.3</version.maven>
+        <version.maven>3.8.1</version.maven>
         <version.assertj>3.19.0</version.assertj>
         <version.jboss-logging>3.4.1.Final</version.jboss-logging>
         <version.jakarta-annotation>1.3.5</version.jakarta-annotation>

--- a/independent-projects/tools/pom.xml
+++ b/independent-projects/tools/pom.xml
@@ -26,7 +26,7 @@
         <maven.compiler.release>11</maven.compiler.release>
 
         <!-- These properties are used by CreateProjectMojo to add the Maven Wrapper -->
-        <proposed-maven-version>3.6.3</proposed-maven-version>
+        <proposed-maven-version>3.8.1</proposed-maven-version>
         <maven-wrapper.version>0.7.7</maven-wrapper.version>
         <gradle-wrapper.version>6.8.3</gradle-wrapper.version>
 
@@ -47,7 +47,7 @@
         <junit.version>5.7.1</junit.version>
         <commons-compress.version>1.20</commons-compress.version>
         <jboss-logging.version>3.4.1.Final</jboss-logging.version>
-        <maven-core.version>3.6.3</maven-core.version>
+        <maven-core.version>3.8.1</maven-core.version>
         <mockito.version>3.9.0</mockito.version>
         <version.surefire.plugin>3.0.0-M5</version.surefire.plugin>
         <nexus-staging-maven-plugin.version>1.6.8</nexus-staging-maven-plugin.version>

--- a/mvnw
+++ b/mvnw
@@ -8,7 +8,7 @@
 # "License"); you may not use this file except in compliance
 # with the License.  You may obtain a copy of the License at
 #
-#    https://www.apache.org/licenses/LICENSE-2.0
+#    http://www.apache.org/licenses/LICENSE-2.0
 #
 # Unless required by applicable law or agreed to in writing,
 # software distributed under the License is distributed on an

--- a/mvnw.cmd
+++ b/mvnw.cmd
@@ -7,7 +7,7 @@
 @REM "License"); you may not use this file except in compliance
 @REM with the License.  You may obtain a copy of the License at
 @REM
-@REM    https://www.apache.org/licenses/LICENSE-2.0
+@REM    http://www.apache.org/licenses/LICENSE-2.0
 @REM
 @REM Unless required by applicable law or agreed to in writing,
 @REM software distributed under the License is distributed on an

--- a/pom.xml
+++ b/pom.xml
@@ -57,7 +57,7 @@
         <postgres.url>jdbc:postgresql:hibernate_orm_test</postgres.url>
 
         <nexus-staging-maven-plugin.version>1.6.8</nexus-staging-maven-plugin.version>
-        <gitflow-incremental-builder.version>3.14.0</gitflow-incremental-builder.version>
+        <gitflow-incremental-builder.version>3.14.1</gitflow-incremental-builder.version>
         <quarkus-platform-bom-plugin.version>0.0.10</quarkus-platform-bom-plugin.version>
 
         <skipDocs>false</skipDocs>

